### ISSUE TITLE
Swig problem  with SimpleMatrix in Debug

### ISSUE THREAD

**Issue by vacary** [siconos/siconos#314]
Testing the example
https://github.com/siconos/siconos-tutorials/blob/master/examples/mechanics/FEM/ImpactingBar/ImpactingBar.py
with a compilation of Siconos in debug raises an assertion:

```
[acary@ohana]$ python ImpactingBar.py
Assertion failed: (own), function SWIG_Python_ConvertPtrAndOwn, file /scratch/Vincent/Build/siconos/debug_testing/wrap/siconos/kernelPYTHON_wrap.cxx, line 2382.
Abort trap: 6
```

It works perfectly well in release. 

It seems that it is related to an issue in SWIG https://github.com/swig/swig/issues/773 but I am not sure.

Did you experience the same problem ? 

I get the same issue with swig4

**Comment by xhub** [comment]
What is on line 2382? If I look at my code generated from swig 4, the only assert would come from
```
             if (newmemory == SWIG_CAST_NEW_MEMORY) {
              assert(own); /* badly formed typemap which will lead to a memory leak - it must set and use own to delete *ptr */
              if (own)
                *own = *own | SWIG_CAST_NEW_MEMORY;
            }
```

which looks like a code smell, there may be a memory leak ... I would be good to catch the signal and print a backtrace.

**Comment by vacary** [comment]
Sorry, to have been a bit fuzzy. It was exactly this assert that is triggered. I will try to produce a smaller example in python that shows the issue.

**Comment by vacary** [comment]
fixed in f05e9b220dcff622678d97146c16b18fc2cc0376
